### DISCUSSION TITLE
Type issue

### DISCUSSION THREAD

**Discussion by Alystrasz** [graphology/graphology#372] [Q&A] [answered]
Hello there!
I'm trying to build a web app that uses graphology, and cannot get it to compile.

Webpack build process points out type errors on lines:
https://github.com/graphology/graphology/blob/7db79282f291f642af07a7cf48c963ac2e6ae04e/src/types/index.d.ts#L43
and
https://github.com/graphology/graphology/blob/7db79282f291f642af07a7cf48c963ac2e6ae04e/src/types/index.d.ts#L63

I could build my app by modifying said types as such (removing properties names):
```typescript
type NodeMergeResult = [string, boolean];
```
```typescript
type EdgeMergeResult = [string, boolean, boolean, boolean];
```

---

Maybe I wrongly setup my typescript config?
Thanks for your help.

**Comment by Yomguithereal** [comment, marked as answer]
@Alystrasz I think this is because named or labeled tuples (the actual name in TS is not clear, even in their own docs) are supported by more recent versions of TS that the one you might be using. Is this acceptable to you to upgrade your version of TS as it should fix the problem you are having?

**Reply by Alystrasz** [reply]
Sure!

Going from `v3.7.5` to `v4.5.5` (and converting whole project accordingly) fixed the issue, so I suppose "named tuples" feature was introduced in Typescript v4.

Thanks!
